### PR TITLE
Wip objectives

### PR DESCRIPTION
**Pull request by zeimedee**
review and leave any questions,comments, suggestions

### CODE REVIEW

File: routes/routes.go [hunk 1]
```diff
@@ -3,33 +3,69 @@ package routes
...
+// 		c.JSON(http.StatusBadRequest, gin.H{"success": false})
+// 		return
+// 	}
+// 	c.JSON(http.StatusOK, gin.H{"success": true, "message": "updated successfully"})
+// }
+
+// func TestDeleteItemHandler(c *gin.Context) {
+// 	var data models.Test
+
+// 	err := database.DeleteItem(data, "id", "1")
```
Comment (blitzblade) [inline routes/routes.go:64]: All these comments, are we removing them or you want to do something with them?

File: main.go [hunk 2]
```diff
@@ -11,14 +11,17 @@ import (
 
 func setupRouter(r *gin.Engine) {
 	r.GET("/", routes.DefaultHandler)
-	r.POST("/test", routes.TestCreateHandler)
-	r.GET("/get", routes.TestGetHandler)
+	// r.POST("/test", routes.TestCreateHandler)
+	// r.GET("/get", routes.TestGetItemsHandler)
+	// r.GET("/getItem", routes.TestGetItemHandler)
+	// r.PUT("/update", routes.TestUpdateItemHandler)
+	// r.DELETE("/delete", routes.TestDeleteItemHandler)
```
Comment (blitzblade) [inline main.go:18]: Comments... Why are they left here?

File: database/database.go [hunk 3]
```diff
@@ -67,27 +67,53 @@ func CreateItem[T any](model T) error {
...
-	if err := db.Find(model).Error; err != nil {
+	if err := db.Where(column+"= ?", value).Find(model).Error; err != nil {
 		return nil, err
 	}
 
 	return model, nil
 }
 
-func GetItem[T any](model T) {}
+func GetItem[T any](model *T, column, value string) (*T, error) {
```
Comment (blitzblade) [inline database/database.go:89]: Have you at least manually tested these generic CRUD methods?

**Review by blitzblade — CHANGES_REQUESTED** [review]
Waiting

File: routes/routes.go (hunk 1 above)
Comment (zeimedee) [reply routes/routes.go:64]: we are removing them

File: database/database.go (hunk 3 above)
Comment (zeimedee) [reply database/database.go:89]: yes, i have manually tested them and they all work

**Pull request closed without merging**